File: frontend/dashboard/types/comfyui-node.ts
export interface ComfyUINode {
  id: number;
  name: string;
  endpoint: string;
  enabled: boolean;
  createdTime: string;
  isAlive: boolean;
  lastCheck: string;
  queue: number;
}

export interface CreateComfyUINodeRequest {
  name: string;
  endpoint: string;
}

export interface UpdateComfyUINodeRequest {
  name?: string;
  endpoint?: string;
  enabled?: boolean;
}

File: dto/comfyuinode.go
package dto

type ComfyUiNodeResponse struct {
	ID          int    `json:"id"`
	Name        string `json:"name"`
	Endpoint    string `json:"endpoint"`
	Enabled     bool   `json:"enabled"`
	CreatedTime string `json:"createdTime"`
	IsAlive     bool   `json:"isAlive"`
	LastCheck   string `json:"lastCheck"`
	Queue       int    `json:"queue"`
}

type CreateComfyUiNodeRequest struct {
	Name     string `json:"name"`
	Endpoint string `json:"endpoint"`
}

type UpdateComfyUiNodeRequest struct {
	Name     *string `json:"name"`
	Endpoint *string `json:"endpoint"`
	Enabled  *bool   `json:"enabled"`
}
